src/mlpack/core/data/split_data.hpp

```diff
@@ -224,6 +228,122 @@ Split(const arma::Mat<T>& input,
                          std::move(testData));
 }
 
+template <class fieldType,
+          class = std::enable_if_t<
+              arma::is_Col<typename fieldType::object_type>::value ||
+              arma::is_Mat_only<typename fieldType::object_type>::value>>
+void Split(fieldType& input, 
+           fieldType& inputLabel,
+           fieldType& trainData,
+           fieldType& trainLabels,
+           fieldType& testData, 
+           fieldType& testLabels,
+           const double testRatio, 
+           const bool shuffleData = true) 
+{
+  const size_t testSize = static_cast<size_t>(input.n_cols * testRatio);
+  const size_t trainSize = input.n_cols - testSize;
+
+  arma::uvec order = arma::linspace<arma::uvec>(0, input.n_cols - 1,
+      input.n_cols);
+  if (shuffleData)
+    order = arma::shuffle(order);
+
+  if (trainSize > 0)
+  {
+    trainLabels.set_size(1, trainSize);
+
+    for (size_t i = 0; i < trainSize; i++)
+       trainData[i] = input(0, order(i));
+
+    // Field type has fixed size so we can't use span and assignment
+    // operator.
```

Comment:
I'm not sure what you mean by this comment---I don't think that the field type has a fixed (compile-time) size?

Reply:
I'm not really sure, maybe @kartikdutt18 could provide some insight..? (I've used that code from [here](https://github.com/mlpack/models/pull/18/files#diff-41f072fe5e54db1663f9b261a99b948aR391))

Reply:
> -I don't think that the field type has a fixed (compile-time) size?

Right they don't, However we can't use insert_cols or something similar corresponding to field type such that we can change it's size from the declared size.

Reply:
I'll delete the comment as well